packages/plugin-legacy/src/index.ts

```diff
@@ -672,7 +675,8 @@ async function buildPolyfillChunk(
           format,
           entryFileNames: rollupOutputOptions.entryFileNames
         }
-      }
+      },
+      terserOptions
```

Comment:
Do we want a new option? I wonder if we could use `buildOptions.terserOptions` here.

Reply:
I thought about that, but there might be a situation in which we want to use different options for each build. In addition, this is more explicit.

Reply:
I also think it isn't a bad idea to have this as a separate option. We may want to add a note in `build.terserOptions` saying that it only applies to terser as a minifier for the main build. I think we may want to remove `build.terserOptions` at one point in the future and push that to a terser plugin too.

Reply:
I'm with @sapphi-red here. I don't think we need to duplicate the options, and the cases that we want different terser options are rare (e.g. we already inherit `minify` option from `buildOptions`)

Reply:
Currently this new option only applies to polyfill chunks (it won't be used for the main chunks). If we want this as an different option, I think the name should include `polyfill`. (e.g. `polyfillChunksTerserOptions` or `polyfillTerserOptions`?)

Also how about having `buildOptions.terserOptions` as a default value of `polyfillChunksTerserOptions`? I guess many won't need to specify a different option.

Reply:
mmm... good points, if `minify` already inherits and we make `buildOptions.terserOptions` acts as the default, then we should just do that for now and wait until we have a proper use case for a new option. You're right.

Reply:
A warning will be shown in the console if `buildOptions.terserOptions` is set but the `minify` option is not set to `terser`. In this case, what should happen? If a user wants to get rid of the message should they set the `polyfillChunksTerserOptions` explicitly or we should check for existence of `plugin-legacy` and remove the warning?

Reply:
I added the `polyfillChunksTerserOptions` option and set the default to `buildOptions.terserOption`.

Reply:
I'd probably lean on a hardcoded `plugin-legacy` check for now, but I'm not sure how others feel about that 😬

Reply:
@bluwy I don't really like that. It's too implicit. If someone is reading the code and sees that `minify` is not set to `terser` and there is a `terserOptions` there, they would think this is redundant! They HAVE to know that `plugin-legacy` is using it which is TOO IMPLICIT!

Reply:
Ah, are you suggesting to add `terserOptions` option that applies to **both polyfill chunks and legacy main chunks**? (This PR's implementation does not do that.)
I think that could be a choice.

Reply:
@sapphi-red We are forcing `terser` only for polyfills. Is that right or I am missing sth?

Reply:
Oh, I saw that we are using `config.build.terserOptions` inside legacy chunks by setting `__vite_force_terser__` to `true`. So this PR is aiming to inherit `terserOptions` for polyfills as well and remove the warning.

On it...

Reply:
Now the warning is gone and I have added another option named as `polyfillChunksTerserOptions` which defaults to `buildOptions.terserOption` for polyfill chunks.

Reply:
I have also added a little documentation to show how to configure `terser` for main and polyfill chunks.